key.go

```diff
@@ -232,6 +233,23 @@ func (k Key) IsTopLevel() bool {
 	return len(k.List()) == 1
 }
 
+// MarshalJSON implements the json.Marshaler interface,
+// keys are represented as JSON strings
+func (k Key) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + k.String() + `"`), nil
+}
+
+// MarshalJSON implements the json.Unmarshaler interface,
+// keys will parse any value specified as a key to a string
+func (k *Key) UnmarshalJSON(data []byte) error {
+	if len(data) < 2 {
+		k.Clean()
+		return errors.New("too short to unmarshal key to json string")
+	}
+	*k = NewKey(string(data[1 : len(data)-1]))
```

Comment:
Whatever escaping we do in the marshal step, we should do the opposite here

Reply:
Again, why not just use the `json` package? Doing our own parsing sounds like a bad idea.